src/Analyzers/CSharp/Analyzers/UseCollectionExpression/UseCollectionExpressionHelpers.cs

```diff
@@ -51,6 +51,18 @@ internal static class UseCollectionExpressionHelpers
         if (originalTypeInfo.ConvertedType is null or IErrorTypeSymbol)
             return false;
 
+        // Special case.  Block "ImmutableArray<T>" when on 7.0 and prior runtimes.  The reason for this is that
+        // ImmutableArray<T> *looks* to be an ok collection-type since it supports the collection-initializer pattern.
+        // However, that pattern does the absolute wrong thing here as it converts to `new ImmutableArray<T>() { X, Y, Z
+        // }` which throws at runtime.
+        var immutableArrayType = compilation.ImmutableArrayOfTType();
+        var collectionBuilderType = compilation.CollectionBuilderAttribute();
+        if (originalTypeInfo.ConvertedType.OriginalDefinition.Equals(immutableArrayType) &&
+            !immutableArrayType.GetAttributes().Any(a => a.AttributeClass?.Equals(collectionBuilderType) is true))
+        {
+            return false;
+        }
```

Comment:
I believe we should modify this to instead disable the analyzer on a type which meets all of the following:

1. Is a value type
2. Does not have an explicitly-defined parameterless constructor (can be detected as [`Constructor.IsImplicitlyDeclared`](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/commit/e2b929699be3f1ca7ebc114420fe72823112c538))
3. Does not have a `[CollectionBuilder]` attribute